pkg/client/fullmesh.go

```diff
@@ -69,17 +72,19 @@ func (fmnsc *FullMeshNetworkServiceClient) Close() error {
 
 func (fmnsc *FullMeshNetworkServiceClient) recv() error {
 	for {
-		networkServiceEndpoint, err := fmnsc.networkServiceDiscoveryStream.Recv()
+		resp, err := fmnsc.networkServiceDiscoveryStream.Recv()
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
+			logrus.Debugf("Full Mesh: Recv err: %v", err)
 			return err
 		}
-
-		if !expirationTimeIsNull(networkServiceEndpoint.ExpirationTime) {
+		networkServiceEndpoint := resp.NetworkServiceEndpoint
+		if !expirationTimeIsNull(networkServiceEndpoint.ExpirationTime) && !resp.Deleted {
```

Comment:
might be possible to remove the check if expiration time is nil, since it has been introduced when the deleted property was not part of the API.
https://github.com/networkservicemesh/api/issues/81